homeassistant/components/overkiz/water_heater_entities/hitachi_dhw.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Hitachi DHW."""
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_HIGH_DEMAND,
+    STATE_OFF,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, TEMP_CELSIUS
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    OverkizCommandParam.STANDARD: STATE_ECO,
+    OverkizCommandParam.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    OverkizCommandParam.STOP: STATE_OFF,
+}
+
+OPERATION_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_OPERATION_MODE.items()}
+
+
+class HitachiDHW(OverkizEntity, WaterHeaterEntity):
+    """Representation of Hitachi DHW."""
+
+    _attr_min_temp = 30.0
+    _attr_max_temp = 70.0
+    _attr_precision = PRECISION_WHOLE
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+    )
+    _attr_operation_list = [*OPERATION_MODE_TO_OVERKIZ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
```

Comment:
Now, you know better our model, can you try again please `value_as_float`?
`self.device.states[OverkizState.CORE_DHW_TEMPERATURE].value_as_float`

Reply:
done there and in others places too